File: 116/[2018AM]有一種選舉叫做智力測驗.cpp
#include <bits/stdc++.h>
using namespace std;

#define miyuki_is_my_wife ios::sync_with_stdio(false), cin.tie(nullptr)

int main()
{
    miyuki_is_my_wife;

    int a1{}, b1{}, a2{}, b2{};
    cin >> a1 >> b1 >> a2 >> b2;
    int sum = a1 + b1;

    if (a2 < a1 && b2 < b1) {
         cout<<"Both players are idiots!\n";
         return 0;
    }

    if (a2 < a1) {
        cout<<"Player A is an idiot!\n";
        return 0;
    }

    if (b2 < b1) {
        cout<<"Player B is an idiot!\n";
        return 0;
    }

    if (a2 == sum && b2 == sum) {
        cout << "Even.\n";
    } else if (a2 == sum) {
        cout << "Player A wins!\n";
    } else if (b2 == sum) {
        cout << "Player B wins!\n";
    } else {
        cout << "Even.\n";
    }

    return 0;
}

File: 116/[2018AM]有一種選舉叫做智力測驗.py
def main():
    a1, b1, a2, b2 = map(int, input().split())
    """
    a1 = int(input())
    b1 = int(input())
    a2 = int(input())
    b2 = int(input())
    """
    
    s = a1 + b1
    if a2 < a1 and b2 < b1:
        print("Both players are idiots!")
        return
        
    if a2 < a1:
        print("Player A is an idiot!")
        return
    
    if b2 < b1:
        print("Player B is an idiot!")
        return    
    
    if a2 == s and b2 == s:
        print("Even.")
    elif a2 == s:
        print("Player A wins!")
    elif b2 == s:
        print("Player B wins!")
    else:
        print("Even.")

if __name__ == "__main__":
    # miyuki is my wife and sister

    main()
